src/app.tsx

```diff
@@ -48,12 +48,36 @@ const SearchReplaceForBlockEditor = () => {
    */
   const onSelection = () => {
     const selectedText = window.getSelection().toString();
+    const modalSelector = '.search-replace-modal';
 
-    if (selectedText) {
+    if (selectedText && !inContainer(modalSelector)) {
       setSearchInput(selectedText);
     }
   };
 
+  /**
```

Comment:
@anandraj346 This PR should only contain changes for this new feature. Pls let's remove this.